src/geometry/manifold/derivation_bundle.lean

```diff
@@ -0,0 +1,163 @@
+/-
+Copyright © 2020 Nicolò Cavalleri. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Nicolò Cavalleri
+-/
+
+import geometry.manifold.algebra.smooth_functions
+import ring_theory.derivation
+
+/-!
+
+# Derivation bundle
+
+In this file we define the derivations at a point of a manifold on the algebra of smooth fuctions.
+Moreover we define the total bundle of derivations (although at the moment it has not been given a
+topology). Finally we define the differential of a function in terms of derivations.
+
+The content of this file is not meant to be regarded as an alternative definition to the current
+tangent bundle but rather as a purely algebraic theory that provides a purely algebraic definition
+of the Lie algebra for a Lie group.
+
+-/
+
+variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+{E : Type*} [normed_group E] [normed_space 𝕜 E]
+{H : Type*} [topological_space H] (I : model_with_corners 𝕜 E H)
+(M : Type*) [topological_space M] [charted_space H M]
+
+open_locale manifold
+
+namespace instances
+def smooth_functions_algebra : algebra 𝕜 C^∞⟮I, M; 𝕜⟯ := infer_instance
+attribute [instance, priority 10000] smooth_functions_algebra
```

Comment:
I believe this is equivalent and more idiomatic:
```suggestion
@[priority 10000]
instance smooth_functions_algebra : algebra 𝕜 C^∞⟮I, M; 𝕜⟯ := infer_instance
```

Also below.

Reply:
Well actually this should be changed to `local attribute` but for some weird reason it does not work, so I am waiting for someone expert in type class inference to have a look at it.